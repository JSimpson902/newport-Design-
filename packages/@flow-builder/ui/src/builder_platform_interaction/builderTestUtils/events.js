export const focusEvent = new FocusEvent('focus', {
    bubbles: false,
    cancelable: false
});

export const focusoutEvent = new FocusEvent('focusout', {
    bubbles: true,
    cancelable: true
});

export const textInputEvent = textInput => {
    return new CustomEvent('textinput', {
        bubbles: true,
        cancelable: true,
        detail: { text: textInput }
    });
};

export const blurEvent = new FocusEvent('blur', {
    bubbles: true,
    cancelable: true
});

export const selectEvent = value => {
    return new CustomEvent('select', {
        bubbles: true,
        cancelable: true,
        detail: { value }
    });
};

export const changeEvent = value => {
    return new CustomEvent('change', {
        bubbles: true,
        cancelable: false,
        detail: { value }
    });
};

export const clickEvent = () => {
    return new CustomEvent('click', {
        bubbles: true,
        cancelable: false
    });
};

export const checkboxChangeEvent = (checked = true) => {
    return new CustomEvent('change', {
        bubbles: true,
        cancelable: false,
        detail: { checked }
    });
};
