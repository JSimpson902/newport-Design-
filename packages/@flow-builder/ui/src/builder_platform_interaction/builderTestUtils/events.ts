// @ts-nocheck
export const focusEvent = new FocusEvent('focus', {
    bubbles: false,
    cancelable: false
});

export const focusoutEvent = new FocusEvent('focusout', {
    bubbles: true,
    cancelable: true
});

export const textInputEvent = textInput => {
    return new CustomEvent('textinput', {
        bubbles: true,
        cancelable: true,
        detail: { text: textInput }
    });
};

export const blurEvent = new FocusEvent('blur', {
    bubbles: true,
    cancelable: true,
    composed: true
});

export const selectEvent = value => {
    return new CustomEvent('select', {
        bubbles: true,
        cancelable: true,
        detail: { value }
    });
};

export const changeEvent = value => {
    return new CustomEvent('change', {
        bubbles: true,
        cancelable: false,
        detail: { value }
    });
};

export const clickEvent = () => {
    return new CustomEvent('click', {
        bubbles: true,
        cancelable: false
    });
};

export const mouseoverEvent = () => {
    return new CustomEvent('mouseover', {
        bubbles: true,
        cancelable: false
    });
};

export const mouseoutEvent = () => {
    return new CustomEvent('mouseout', {
        bubbles: true,
        cancelable: false
    });
};

export const checkboxChangeEvent = (checked = true) => {
    return new CustomEvent('change', {
        bubbles: true,
        cancelable: false,
        detail: { checked }
    });
};

/**
 * Remove event (used on pill for instance)
 */
export const removeEvent = (): CustomEvent => {
    return new CustomEvent('remove', {
        bubbles: true,
        cancelable: false
    });
};

export const lightningRadioGroupChangeEvent = newValue => {
    return new CustomEvent('change', {
        detail: {
            newValue
        },
        composed: true,
        bubbles: true,
        cancelable: true
    });
};
