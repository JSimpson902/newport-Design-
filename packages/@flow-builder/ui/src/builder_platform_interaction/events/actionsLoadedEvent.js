/**
 * Used by actions to indicate that it's action has changed, includes action type and number of actions of that type
 */
const eventName = 'actionsloaded';

export class ActionsLoadedEvent {
    constructor(value = null, number = null) {
        return new CustomEvent(eventName, {
            cancelable: false,
            composed: true,
            bubbles: true,
            detail: {
                value,
                number
            }
        });
    }

    static EVENT_NAME = eventName;
}
