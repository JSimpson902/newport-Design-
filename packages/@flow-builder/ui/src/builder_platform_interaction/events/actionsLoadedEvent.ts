/**
 * Used by actions to indicate that it's action has changed, includes action type and number of actions of that type
 */
const eventName = 'actionsloaded';

export class ActionsLoadedEvent extends CustomEvent<any> {
    constructor(value = null, number = null) {
        super(eventName, {
            cancelable: false,
            composed: true,
            bubbles: true,
            detail: {
                value,
                number
            }
        });
    }

    static EVENT_NAME = eventName;
}
