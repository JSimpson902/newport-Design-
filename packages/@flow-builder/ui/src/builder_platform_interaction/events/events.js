/*
 * Contains all custom event classes
 */
export { PropertyChangedEvent } from './propertyChangedEvent';
export { FetchMenuDataEvent } from './fetchMenuDataEvent';
export { ValueChangedEvent } from './valueChangedEvent';
export { AddListItemEvent } from './addListItemEvent';
export { UpdateListItemEvent } from './updateListItemEvent';
export { DeleteListItemEvent } from './deleteListItemEvent';
export { RowContentsChangedEvent } from './rowContentsChangedEvent';