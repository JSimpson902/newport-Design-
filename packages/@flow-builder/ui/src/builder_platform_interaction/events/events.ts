/*
 * Contains all custom event classes
 */

export { ActionsLoadedEvent } from './actionsLoadedEvent';
export { AddConditionEvent } from './addConditionEvent';
export { AddListItemEvent } from './addListItemEvent';
export { AddNodeEvent } from './addNodeEvent';
export { AddRecordFieldAssignmentEvent } from './addRecordFieldAssignmentEvent';
export { AddRecordFilterEvent } from './addRecordFilterEvent';
export { AddRecordLookupFieldEvent } from './addRecordLookupFieldEvent';
export { AddSortOptionItemEvent } from './addSortOptionItemEvent';
export { ArrowKeyDownEvent } from './arrowKeyDownEvent';
export { CannotRetrieveActionsEvent } from './cannotRetrieveActionsEvent';
export { CannotRetrieveCalloutParametersEvent } from './cannotRetrieveCalloutParametersEvent';
export { CannotRetrieveTemplatesEvent } from './cannotRetrieveTemplatesEvent';
export { AddConnectionEvent } from './canvasEvents/addConnectionEvent';
export { CANVAS_EVENT, MARQUEE_ACTION, ZOOM_ACTION } from './canvasEvents/canvasEvents';
export { CanvasMouseUpEvent } from './canvasEvents/canvasMouseUpEvent';
export { ConnectorSelectedEvent } from './canvasEvents/connectorSelectedEvent';
export { CopySingleElementEvent } from './canvasEvents/copySingleElementEvent';
export { DragNodeEvent } from './canvasEvents/dragNodeEvent';
export { DragNodeStopEvent } from './canvasEvents/dragNodeStopEvent';
export { MarqueeSelectEvent } from './canvasEvents/marqueeSelectEvent';
export { NodeMouseDownEvent } from './canvasEvents/nodeMouseDownEvent';
export { SelectNodeEvent } from './canvasEvents/selectNodeEvent';
export { ToggleMarqueeOnEvent } from './canvasEvents/zoomPanelEvent/toggleMarqueeOnEvent';
export { ClickToZoomEvent } from './canvasEvents/zoomPanelEvent/zoomEvent';
export { ClosePropertyEditorEvent } from './closePropertyEditorEvent';
export { CollectionReferenceChangedEvent } from './collectionReferenceChangedEvent';
export { ComboboxStateChangedEvent } from './comboboxStateChangedEvent';
export { ConfigurationEditorChangeEvent } from './configurationEditorChangeEvent';
export { ConfigurationEditorPropertyDeleteEvent } from './configurationEditorPropertyDeleteEvent';
export { ConfigurationEditorTypeMappingChangeEvent } from './configurationEditorTypeMappingChangeEvent';
export { CreateEntryConditionsEvent } from './createEntryConditionsEvent';
export { DebugPanelFilterEvent } from './debugPanelEvents/debugPanelFilterEvent';
export { ResumeDebugFlowEvent } from './debugPanelEvents/resumeDebugFlowEvent';
export { DeleteAllConditionsEvent } from './deleteAllConditionsEvent';
export { DeleteConditionEvent } from './deleteConditionEvent';
export { DeleteListItemEvent } from './deleteListItemEvent';
export { DeleteOrchestrationActionEvent } from './deleteOrchestrationActionEvent';
export { DeleteOutcomeEvent } from './deleteOutcomeEvent';
export { DeleteParameterItemEvent } from './deleteParameterItemEvent';
export { DeleteRecordFieldAssignmentEvent } from './deleteRecordFieldAssignmentEvent';
export { DeleteRecordFilterEvent } from './deleteRecordFilterEvent';
export { DeleteRecordLookupFieldEvent } from './deleteRecordLookupFieldEvent';
export { DeleteResourceEvent } from './deleteResourceEvent';
export { DeleteScheduledPathEvent } from './deleteScheduledPathEvent';
export { DeleteSortOptionItemEvent } from './deleteSortOptionItemEvent';
export { DeleteWaitEventEvent } from './deleteWaitEventEvent';
export { DummyPreviewModeEvent } from './dummyPreviewModeEvent';
export { DynamicTypeMappingChangeEvent } from './dynamicTypeMappingChangeEvent';
export { EditListItemEvent } from './editListItemEvent';
export { EditMergeFieldPillEvent } from './editMergeFieldPillEvent';
export { AddElementEvent } from './elementEvents/addElementEvent';
export { AddNonCanvasElementEvent } from './elementEvents/addNonCanvasElementEvent';
export { DeleteElementEvent, DeleteElementEventDetail } from './elementEvents/deleteElementEvent';
export { EditElementEvent } from './elementEvents/editElementEvent';
export { ToggleElementEvent } from './elementEvents/toggleElementEvent';
export { ExecuteWhenOptionChangedEvent } from './executeWhenOptionChangedEvent';
export { FetchMenuDataEvent } from './fetchMenuDataEvent';
export { FilterMatchesEvent } from './filterMatchesEvent';
export { InputsOnNextNavToAssocScrnChangeEvent } from './inputsOnNextNavToAssocScrnChangeEvent';
export { ItemSelectedEvent } from './itemSelectedEvent';
export { LegalNoticeDismissedEvent } from './legalNoticeDismissedEvent';
export { ListItemInteractionEvent } from './listItemInteractionEvent';
export { LoopCollectionChangedEvent } from './loopCollectionChangedEvent';
export { ManuallyAssignVariablesChangedEvent } from './manuallyAssignVariablesChangedEvent';
export { NewResourceEvent } from './newResourceEvent';
export { NumberRecordToStoreChangedEvent } from './numberRecordToStoreChangedEvent';
export { OpenSubflowEvent } from './openSubflowEvent';
export { ORCHESTRATED_ACTION_CATEGORY } from './orchestratedActionCategory';
export { OrchestrationActionValueChangedEvent } from './orchestrationActionValueChangedEvent';
export { OrchestrationAssigneeChangedEvent } from './orchestrationAssigneeChangedEvent';
export { LocatorIconClickedEvent } from './paletteEvents/locatorIconClickedEvent';
export { PaletteItemChevronClickedEvent } from './paletteEvents/paletteItemChevronClickedEvent';
export { PaletteItemClickedEvent } from './paletteEvents/paletteItemClickedEvent';
export { PrepopulateMapItemsEvent } from './prepopulateMapItemsEvent';
export { ProcessTypeSelectedEvent } from './processTypeSelectedEvent';
export { PropertyChangedEvent } from './propertyChangedEvent';
export { PropertyEditorWarningEvent } from './propertyEditorWarningEvent';
export { RecordStoreOptionChangedEvent } from './recordStoreOptionChangedEvent';
export { RemoveMergeFieldPillEvent } from './removeMergeFieldPillEvent';
export { ReorderListEvent } from './reorderListEvent';
export { RequiresAsyncProcessingChangedEvent } from './requiresAsyncProcessingChangedEvent';
export { RichTextPlainTextSwitchChangedEvent } from './richTextPlainTextSwitchChangedEvent';
export { RowContentsChangedEvent } from './rowContentsChangedEvent';
export * from './screenEditorEvents';
export { SetPropertyEditorTitleEvent } from './setPropertyEditorTitleEvent';
export { ShowResourceDetailsEvent } from './showResourceDetailsEvent';
export { SObjectReferenceChangedEvent } from './sObjectReferenceChangedEvent';
export { TemplateChangedEvent } from './templateChangedEvent';
export { UpdateTestAssertionEvent } from './testEditorEvents/UpdateTestAssertionEvent';
export { TextChangedEvent } from './textChangedEvent';
export { AddToFlowTestEvent } from './toolbarEvents/addToFlowTestEvent';
export { CopyOnCanvasEvent } from './toolbarEvents/copyOnCanvasEvent';
export { DebugFlowEvent } from './toolbarEvents/debugFlowEvent';
export { DuplicateEvent } from './toolbarEvents/duplicateEvent';
export { EditFlowEvent } from './toolbarEvents/editFlowEvent';
export { EditFlowPropertiesEvent } from './toolbarEvents/editFlowPropertiesEvent';
export { EditTestEvent } from './toolbarEvents/editTestEvent';
export { NewDebugFlowEvent } from './toolbarEvents/newDebugFlowEvent';
export { RedoEvent } from './toolbarEvents/redoEvent';
export { RestartDebugFlowEvent } from './toolbarEvents/restartDebugFlowEvent';
export { RunFlowEvent } from './toolbarEvents/runFlowEvent';
export { SaveFlowEvent } from './toolbarEvents/saveFlowEvent';
export { ToggleCanvasModeEvent } from './toolbarEvents/toggleCanvasModeEvent';
export { ToggleFlowStatusEvent } from './toolbarEvents/toggleFlowStatusEvent';
export { ToggleSelectionModeEvent } from './toolbarEvents/toggleSelectionModeEvent';
export { ToolbarFocusOutEvent } from './toolbarEvents/toolbarFocusOutEvent';
export { UndoEvent } from './toolbarEvents/undoEvent';
export { ViewAllTestsEvent } from './toolbarEvents/viewAllTestsEvent';
export { UpdateCollectionProcessorEvent } from './updateCollectionProcessorEvent';
export { UpdateConditionEvent } from './updateConditionEvent';
export { UpdateConditionLogicEvent } from './updateConditionLogicEvent';
export { UpdateEntryExitCriteriaEvent } from './updateEntryExitCriteriaEvent';
export { UpdateListItemEvent } from './updateListItemEvent';
export { UpdateNodeEvent } from './updateNodeEvent';
export { UpdateParameterItemEvent } from './updateParameterItemEvent';
export { UpdateRecordFieldAssignmentEvent } from './updateRecordFieldAssignmentEvent';
export { UpdateRecordFilterEvent } from './updateRecordFilterEvent';
export { UpdateRecordLookupFieldEvent } from './updateRecordLookupFieldEvent';
export { UpdateSortCollectionOutputEvent } from './updateSortCollectionOutputEvent';
export { UpdateSortOptionItemEvent } from './updateSortOptionItemEvent';
export { UpdateWaitEventEventTypeEvent } from './updateWaitEventEventTypeEvent';
export { ValidationRuleChangedEvent } from './validationRuleChangedEvent';
export { ValueChangedEvent } from './valueChangedEvent';
export { VariableAndFieldMappingChangedEvent } from './variableAndFieldMappingChangedEvent';
export { VisualPickerItemChangedEvent } from './visualPickerItemChangedEvent';
export { VisualPickerListChangedEvent } from './visualPickerListChangedEvent';
export { WaitEventAddParameterEvent } from './waitEventAddParameterEvent';
export { WaitEventDeleteParameterEvent } from './waitEventDeleteParameterEvent';
export { WaitEventParameterChangedEvent } from './waitEventParameterChangedEvent';
export { WaitEventPropertyChangedEvent } from './waitEventPropertyChangedEvent';
