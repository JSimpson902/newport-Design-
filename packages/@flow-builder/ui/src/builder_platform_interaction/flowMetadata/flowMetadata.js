export const FLOW_PROCESS_TYPE = {
    AUTO_LAUNCHED_FLOW : 'AutoLaunchedFlow',
    FLOW : 'Flow'
};

export const ELEMENT_TYPE = {
    ACTION_CALL: 'ACTION_CALL',
    ASSIGNMENT: 'ASSIGNMENT',
    VARIABLE: 'VARIABLE',
    DECISION: 'DECISION',
    DECISION_WITH_MODIFIED_AND_DELETED_OUTCOMES: 'DECISION_WITH_MODIFIED_AND_DELETED_OUTCOMES',
    APEX_PLUGIN_CALL: 'APEX_PLUGIN_CALL',
    APEX_CALL: 'APEX_CALL',
    EMAIL_ALERT: 'EMAIL_ALERT',
    CHOICE: 'CHOICE',
    CONSTANT: 'CONSTANT',
    DYNAMIC_CHOICE_SET: 'DYNAMIC_CHOICE_SET',
    FORMULA: 'FORMULA',
    STAGE: 'STAGE',
    TEXT_TEMPLATE: 'TEXT_TEMPLATE',
    LOOP: 'LOOP',
    OUTCOME: 'OUTCOME',
    RECORD_CREATE: 'RECORD_CREATE',
    RECORD_DELETE: 'RECORD_DELETE',
    RECORD_LOOKUP: 'RECORD_LOOKUP',
    RECORD_UPDATE: 'RECORD_UPDATE',
    SCREEN: 'SCREEN',
    STEP: 'STEP',
    SUBFLOW: 'SUBFLOW',
    START_ELEMENT: 'START_ELEMENT',
    WAIT: 'WAIT',
    FLOW_PROPERTIES: 'FLOW_PROPERTIES',
    DEFAULT: 'defaultElement'
};

export const CONNECTOR_TYPE = {
    REGULAR: 'REGULAR',
    FAULT: 'FAULT',
    DEFAULT: 'DEFAULT',
    START: 'START',
    LOOP_NEXT: 'LOOP_NEXT',
    LOOP_END: 'LOOP_END'
};

export const SUB_ELEMENT_TYPE = {
    ASSIGNMENT_ITEM: 'ASSIGNMENT_ITEM',
    CONDITION: 'CONDITION',
    RECORD_LOOKUP_FILTER_ITEM: 'RECORD_LOOKUP_FILTER_ITEM',
    RECORD_LOOKUP_FIELD: 'RECORD_LOOKUP_FIELD',
};

export const ACTION_TYPE = {
    APEX: 'apex',
    EMAIL_ALERT: 'emailAlert',
    QUICK_ACTION: 'quickAction',
    FLOW: 'flow',
    COMPONENT: 'component'
};

export const METADATA_KEY = {
    ACTION_CALLS: 'actionCalls',
    APEX_PLUGIN_CALLS: 'apexPluginCalls',
    ASSIGNMENTS: 'assignments',
    DECISIONS: 'decisions',
    WAITS: 'waits',
    FORMULAS: 'formulas',
    VARIABLES: 'variables',
    CONSTANTS: 'constants',
    SUBFLOWS: 'subflows',
    RECORD_CREATE: 'recordCreates',
    RECORD_LOOKUP: 'recordLookups',
    RECORD_DELETE: 'recordDeletes',
    RECORD_UPDATE: 'recordUpdates',
    LOOPS: 'loops',
    SCREENS: 'screens'
};

/**
 * AND and OR are flow values which are used by the backend as well as being as used
 * on the frontend (combobox values).  CUSTOM_LOGIC is used only by the frontend as a
 * combobox value
 * @type {{AND: string, OR: string, ALWAYS: string, CUSTOM_LOGIC: string}}
 */
export const CONDITION_LOGIC = {
    AND: 'and',
    OR: 'or',
    CUSTOM_LOGIC: 'custom_logic',
    NO_CONDITIONS: 'no_conditions'
};

/**
 * The top level process metadata values to save with each flow. This includes the BuilderType
 * which helps us identify flow versions created by the Lightning Flow Builder.
 */
export const PROCESS_METADATA_VALUES = [
    {
        name: 'BuilderType',
        value: {
            stringValue: 'LightningFlowBuilder'
        }
    }
];

/**
 * List of Flow metadata fields that can be references to other Flow elements
 * i.e. fields whose values need to be converted from dev names to GUIDs when loading the flow into the UI client and vice versa.
 */
export const REFERENCE_FIELDS = new Set([
    'elementReference',
    'targetReference',
    'assignToReference',
    'assignNextValueToReference',
    'collectionReference',
    'inputReference',
    'outputReference',
    'assignRecordIdToReference',
    'leftValueReference',
    'outcomeReference',
    'childReference',
    'leftHandSide',
    'defaultValue',
    'source',
    'target',
    'rightHandSideGuid'
]);

/**
 * List of Flow metadata fields that can contain {!reference} tags referencing flow elements
 * These would also need to be replaced
 */
export const TEMPLATE_FIELDS = new Set([
    // TODO fill out more values and confirm that all FEROV stringValue fields allow this behaviour
    'stringValue', // field of a ferov
    'rightHandSide', // field of a ferov
    'expression',  // represents body of a formula
    'text',        // represents body of a text template
    'fieldText',   // body of screen field of type Display Text
    'helpText',     // help text for a screen or screen field
    'interviewLabel' // interview label for the flow properties
]);

export const EXPRESSION_RE = /\{!([^}]+)\}/g;
