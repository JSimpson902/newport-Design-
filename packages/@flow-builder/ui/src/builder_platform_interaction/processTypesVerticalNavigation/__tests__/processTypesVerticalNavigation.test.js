import { createElement } from 'lwc';
import { getShadowRoot } from 'lwc-test-utils';
import ProcessTypesVerticalNavigation from '../processTypesVerticalNavigation';
import { ALL_PROCESS_TYPE, PROCESS_TYPES_ICONS, PROCESS_TYPE_DEFAULT_ICON} from '../processTypesVerticalNavigationUtil';
import { FLOW_PROCESS_TYPE } from "builder_platform_interaction/flowMetadata";
import { ProcessTypeSelectedEvent } from 'builder_platform_interaction/events';
import { MOCK_PROCESS_TYPES, MOCK_ALL_PROCESS_TYPES } from "mock/processTypesData";

const createComponentUnderTest = (processTypes = MOCK_ALL_PROCESS_TYPES) => {
    const el = createElement('builder_platform_interaction-process-types-vertical-navigation', { is: ProcessTypesVerticalNavigation });
    Object.assign(el, {processTypes});
    document.body.appendChild(el);
    return el;
};

const SELECTORS = {
    VERTICAL_NAVIGATION_ITEM_ICON: 'lightning-vertical-navigation-item-icon',
    VERTICAL_NAVIGATION_ITEM_ICON_ANCHOR : 'a',
};

const getAllVerticalNavigationItemIcons = processTypesVerticalNavigation => getShadowRoot(processTypesVerticalNavigation).querySelectorAll(SELECTORS.VERTICAL_NAVIGATION_ITEM_ICON);
const getProcessTypeAnchor = processTypesVerticalNavigationItemIcon => getShadowRoot(processTypesVerticalNavigationItemIcon).querySelector(SELECTORS.VERTICAL_NAVIGATION_ITEM_ICON_ANCHOR);

describe('process-types-vertical-navigation', () => {
        let processTypesVerticalNavigation;
        beforeEach(() => {
            processTypesVerticalNavigation = createComponentUnderTest();
        });

        describe('UI rendering (snapshot)', () => {
            test('label and truncation (via CSS)', () => {
                expect(processTypesVerticalNavigation).toMatchSnapshot();
            });
        });
        describe('Process types details', () => {
            test('by default "all" entry selected', () => {
                expect(processTypesVerticalNavigation.selectedProcessType).toBe(ALL_PROCESS_TYPE.name);
            });

            test('number of "featured" process types', () => {
                expect(processTypesVerticalNavigation.featuredProcessTypes).toHaveLength(MOCK_PROCESS_TYPES.FEATURED.length + 1); // 'ALL' entry included
            });

            test('number of "other" process types', () => {
                expect(processTypesVerticalNavigation.otherProcessTypes).toHaveLength(MOCK_PROCESS_TYPES.OTHERS.length);
            });

            test('"All" entry in first place (among featured process types) (API)', () => {
                expect(processTypesVerticalNavigation.featuredProcessTypes[0]).toMatchObject({name:ALL_PROCESS_TYPE.name});
            });

            test('details of "featured" process types ("all" included) (API)', () => {
                expect(processTypesVerticalNavigation.featuredProcessTypes).toEqual(expect.arrayContaining([
                    expect.objectContaining({ label: 'FlowBuilderProcessTypesVerticalNavigation.all', name: ALL_PROCESS_TYPE.name, iconName: PROCESS_TYPES_ICONS.FEATURED.get(ALL_PROCESS_TYPE.name)}),
                    expect.objectContaining({ label: 'Autolaunched Flow', name: FLOW_PROCESS_TYPE.AUTO_LAUNCHED_FLOW, iconName: PROCESS_TYPES_ICONS.FEATURED.get(FLOW_PROCESS_TYPE.AUTO_LAUNCHED_FLOW)}),
                    expect.objectContaining({ label: 'Screen Flow', name: FLOW_PROCESS_TYPE.FLOW, iconName: PROCESS_TYPES_ICONS.FEATURED.get(FLOW_PROCESS_TYPE.FLOW)})]));
            });

            test('details of "other" process types (API)', () => {
                expect(processTypesVerticalNavigation.otherProcessTypes).toEqual(expect.arrayContaining([
                    expect.objectContaining({ label: 'Checkout Flow', name: FLOW_PROCESS_TYPE.CHECKOUT_FLOW, iconName: PROCESS_TYPES_ICONS.OTHERS.get(FLOW_PROCESS_TYPE.CHECKOUT_FLOW)}),
                    expect.objectContaining({ label: 'Contact Request Flow', name: FLOW_PROCESS_TYPE.CONTACT_REQUEST_FLOW, iconName: PROCESS_TYPES_ICONS.OTHERS.get(FLOW_PROCESS_TYPE.CONTACT_REQUEST_FLOW)}),
                    expect.objectContaining({ label: 'Embedded Appointment Management Flow', name: FLOW_PROCESS_TYPE.FIELD_SERVICE_WEB, iconName: PROCESS_TYPES_ICONS.OTHERS.get(FLOW_PROCESS_TYPE.FIELD_SERVICE_WEB)}),
                    expect.objectContaining({ label: 'Field Service Mobile Flow', name: FLOW_PROCESS_TYPE.FIELD_SERVICE_MOBILE, iconName: PROCESS_TYPES_ICONS.OTHERS.get(FLOW_PROCESS_TYPE.FIELD_SERVICE_MOBILE)}),
                    expect.objectContaining({ label: 'User Provisioning Flow', name: FLOW_PROCESS_TYPE.USER_PROVISIONING_FLOW, iconName: PROCESS_TYPES_ICONS.OTHERS.get(FLOW_PROCESS_TYPE.USER_PROVISIONING_FLOW)}),
                    expect.objectContaining({ label: 'Well no icon yet', name: 'WeDoNotKnowYou', iconName: PROCESS_TYPE_DEFAULT_ICON})]));
            });

            test('process types name', () => {
                const processTypesVerticalNavigationItemIcons = getAllVerticalNavigationItemIcons(processTypesVerticalNavigation);
                const joinedProcessTypesNames = Array.from(processTypesVerticalNavigationItemIcons).map(itemIcon => {
                    return itemIcon.name;
                }).join('');
                expect(joinedProcessTypesNames).toBe(ALL_PROCESS_TYPE.name + MOCK_ALL_PROCESS_TYPES.map(processType => {
                    return processType.name;
                }).join(''));
            });
        });

        describe('Events', () => {
            test('should fire "ProcessTypeSelectedEvent" when selecting a process type with process type name', async () => {
                const eventCallback = jest.fn();
                processTypesVerticalNavigation.addEventListener(ProcessTypeSelectedEvent.EVENT_NAME, eventCallback);
                const processTypesVerticalNavigationItemIcon = getAllVerticalNavigationItemIcons(processTypesVerticalNavigation)[1];
                const autolaunchedFlowAnchor = getProcessTypeAnchor(processTypesVerticalNavigationItemIcon);
                autolaunchedFlowAnchor.dispatchEvent(new Event('click'));
                await Promise.resolve();
                expect(eventCallback).toHaveBeenCalled();
                expect(eventCallback.mock.calls[0][0].detail).toEqual({name: FLOW_PROCESS_TYPE.AUTO_LAUNCHED_FLOW});
            });
    });
});