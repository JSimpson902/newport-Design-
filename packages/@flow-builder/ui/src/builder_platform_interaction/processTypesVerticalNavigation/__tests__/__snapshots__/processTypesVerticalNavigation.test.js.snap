// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`process-types-vertical-navigation UI rendering (snapshot) label and truncation (via CSS) 1`] = `
<builder_platform_interaction-process-types-vertical-navigation>
  #shadow-root(open)
    <div
      aria-hidden="false"
      class="navigation-container slds-size_small slds-panel slds-panel_drawer slds-border_right slds-panel_docked-left slds-grid slds-grid_vertical"
    >
      <div
        class="slds-scrollable_y slds-grid slds-grid_vertical slds-grow"
      >
        <lightning-vertical-navigation>
          #shadow-root(open)
            <slot />
          <lightning-vertical-navigation-item-icon
            class="slds-truncate"
            title="FlowBuilderProcessTypesVerticalNavigation.all"
          >
            #shadow-root(open)
              <a
                class="slds-nav-vertical__action"
                style="display:inline-block;"
              >
                <lightning-icon>
                  #shadow-root(open)
                </lightning-icon>
                FlowBuilderProcessTypesVerticalNavigation.all
              </a>
          </lightning-vertical-navigation-item-icon>
          <lightning-vertical-navigation-item-icon
            class="slds-truncate"
            title="Autolaunched Flow"
          >
            #shadow-root(open)
              <a
                class="slds-nav-vertical__action"
                style="display:inline-block;"
              >
                <lightning-icon>
                  #shadow-root(open)
                </lightning-icon>
                Autolaunched Flow
              </a>
          </lightning-vertical-navigation-item-icon>
          <lightning-vertical-navigation-item-icon
            class="slds-truncate"
            title="Screen Flow"
          >
            #shadow-root(open)
              <a
                class="slds-nav-vertical__action"
                style="display:inline-block;"
              >
                <lightning-icon>
                  #shadow-root(open)
                </lightning-icon>
                Screen Flow
              </a>
          </lightning-vertical-navigation-item-icon>
          <lightning-vertical-navigation-overflow>
            #shadow-root(open)
              <slot />
            <div
              class="slds-p-left_small"
            >
              <lightning-vertical-navigation-item-icon
                class="slds-truncate"
                title="Checkout Flow"
              >
                #shadow-root(open)
                  <a
                    class="slds-nav-vertical__action"
                    style="display:inline-block;"
                  >
                    <lightning-icon>
                      #shadow-root(open)
                    </lightning-icon>
                    Checkout Flow
                  </a>
              </lightning-vertical-navigation-item-icon>
              <lightning-vertical-navigation-item-icon
                class="slds-truncate"
                title="Contact Request Flow"
              >
                #shadow-root(open)
                  <a
                    class="slds-nav-vertical__action"
                    style="display:inline-block;"
                  >
                    <lightning-icon>
                      #shadow-root(open)
                    </lightning-icon>
                    Contact Request Flow
                  </a>
              </lightning-vertical-navigation-item-icon>
              <lightning-vertical-navigation-item-icon
                class="slds-truncate"
                title="Embedded Appointment Management Flow"
              >
                #shadow-root(open)
                  <a
                    class="slds-nav-vertical__action"
                    style="display:inline-block;"
                  >
                    <lightning-icon>
                      #shadow-root(open)
                    </lightning-icon>
                    Embedded Appointment Management Flow
                  </a>
              </lightning-vertical-navigation-item-icon>
              <lightning-vertical-navigation-item-icon
                class="slds-truncate"
                title="Field Service Mobile Flow"
              >
                #shadow-root(open)
                  <a
                    class="slds-nav-vertical__action"
                    style="display:inline-block;"
                  >
                    <lightning-icon>
                      #shadow-root(open)
                    </lightning-icon>
                    Field Service Mobile Flow
                  </a>
              </lightning-vertical-navigation-item-icon>
              <lightning-vertical-navigation-item-icon
                class="slds-truncate"
                title="User Provisioning Flow"
              >
                #shadow-root(open)
                  <a
                    class="slds-nav-vertical__action"
                    style="display:inline-block;"
                  >
                    <lightning-icon>
                      #shadow-root(open)
                    </lightning-icon>
                    User Provisioning Flow
                  </a>
              </lightning-vertical-navigation-item-icon>
              <lightning-vertical-navigation-item-icon
                class="slds-truncate"
                title="Well no icon yet"
              >
                #shadow-root(open)
                  <a
                    class="slds-nav-vertical__action"
                    style="display:inline-block;"
                  >
                    <lightning-icon>
                      #shadow-root(open)
                    </lightning-icon>
                    Well no icon yet
                  </a>
              </lightning-vertical-navigation-item-icon>
            </div>
          </lightning-vertical-navigation-overflow>
        </lightning-vertical-navigation>
      </div>
    </div>
</builder_platform_interaction-process-types-vertical-navigation>
`;
